src/main/java/tree/ImpTargetTreeNode.kt

```diff
@@ -27,6 +29,40 @@ class ImpTargetTreeNode(
         return null
     }
 
+    override fun matchedNodesString(selectedNodes: List<TargetTreeNode>): String {
+        return matchedNodesString(selectedNodes, 0, Int.MAX_VALUE).matchedString
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    fun matchedNodesString(selectedNodes: List<TargetTreeNode>, initialIndex: Int, rightSiblingId: Int): MatchedResponse {
+        val iAmIn = selectedNodes[initialIndex].id == id
+        var str = ""
+        var currentIndex = initialIndex
+        if (iAmIn)
+            currentIndex++
+        while (currentIndex < selectedNodes.size && selectedNodes[currentIndex].id < rightSiblingId) {
+            var goalInd = (children as List<ImpTargetTreeNode>).binarySearch(ImpTargetTreeNode("", "", selectedNodes[currentIndex].id), ImpTargetTreeNodeComparator())
+
+            if (goalInd < 0 || (goalInd == 0 && selectedNodes[currentIndex].id != children[0].id)) {
+                goalInd *= -1
+                goalInd -= 1 // <---------- This might not be needed in the future when searching for siblings but cannot be tested right now
```

Comment:
This is a line that might be necessary or might break the code, however this cannot be tested right now and needs the changes made in PR #41 to test wether it needs to be removed. Just leaving that note here so everyone can read it.